pkg/operator/controller/controller_dns_configmap_test.go

```diff
@@ -1141,6 +1141,78 @@ func Test_coreDNSCache(t *testing.T) {
 	}
 }
 
+func TestDesiredDNSConfigmapDNSNameResolverEnabled(t *testing.T) {
+	testCases := []struct {
+		name             string
+		dns              *operatorv1.DNS
+		expectedCoreFile string
+		expectedError    error
+	}{
+		{
+			name: "Check if Corefile is rendered correctly with ocp_dnsnameresolver plugin",
+			dns: &operatorv1.DNS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: DefaultDNSController,
+				},
+				Spec: operatorv1.DNSSpec{
+					Servers: []operatorv1.Server{
+						{
+							Name:  "foo",
+							Zones: []string{"foo.com"},
+							ForwardPlugin: operatorv1.ForwardPlugin{
+								Upstreams: []string{"1.1.1.1", "2.2.2.2:5353"},
+								Policy:    operatorv1.RoundRobinForwardingPolicy,
+							},
+						},
+						{
+							Name:  "bar",
+							Zones: []string{"bar.com", "example.com"},
+							ForwardPlugin: operatorv1.ForwardPlugin{
+								Upstreams: []string{"3.3.3.3"},
+								Policy:    operatorv1.RandomForwardingPolicy,
+							},
+						},
+						{
+							Name:  "fizz",
+							Zones: []string{"fizz.com"},
+							ForwardPlugin: operatorv1.ForwardPlugin{
+								Upstreams: []string{"5.5.5.5", "6.6.6.6"},
+								Policy:    operatorv1.SequentialForwardingPolicy,
+							},
+						},
+						{
+							Name:  "buzz",
+							Zones: []string{"buzz.com", "example.buzz.com"},
+							ForwardPlugin: operatorv1.ForwardPlugin{
+								Upstreams: []string{"4.4.4.4"},
+							},
+						},
+					},
+				},
+			},
```

Comment:
Do we really need that many servers to check the plugin is added?

Reply:
I have added 4 servers to show that the ocp_dnsnameresolver plugin gets added to each server config.

Reply:
Wouldn't 2 servers be sufficient then?

Reply:
Ok. I'll reduce the number of servers to 2.